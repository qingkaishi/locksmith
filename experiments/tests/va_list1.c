#include <stdarg.h>
#include <pthread.h>

struct hist_data {
   int req ;
};
void Log(char *fmt  , ...) ;
void resume_get() ;
void *signal_waiter(void *arg ) ;
int *req  ;
pthread_t hthread  ;
struct thread_data *wthread  ;
int main(int argc , char **argv ) {
  struct hist_data h ;

  req = (int *)calloc(1U, sizeof(int ));
  pthread_create(&hthread, 0, & signal_waiter, 0);
  Log((char *)"Attempting to read log file %s for resuming download job...", "A");
  resume_get();
  return (0);
}
void resume_get() {
  int i;
  memcpy(req, &i, 0);
}
void Log(char *fmt  , ...) 
{ va_list ap ;
  va_start(ap, fmt);
  va_end(ap);
}
void *signal_waiter(void *arg ) 
{ 
  int i;
  memcpy(&i, req, 0);
  return ((void *)0);
}
