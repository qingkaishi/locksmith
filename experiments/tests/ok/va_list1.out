************************* STARTING *************************

functions declared and used but not defined:
  __builtin_va_end
shared: alloc:va_list1.c:16

Warning: Possible data race: alloc:va_list1.c:16 is not protected!
 references:
  dereference of req:va_list1.c:10 at va_list1.c:24
    alloc:va_list1.c:16 => req:va_list1.c:10
  locks acquired:
    <empty>
  in: main at va_list1.c:13  -> va_list1.c:19

  dereference of req:va_list1.c:10 at va_list1.c:34
    alloc:va_list1.c:16 => req:va_list1.c:10
  locks acquired:
    <empty>
  in: main at va_list1.c:13 


*************************** DONE ***************************
